daisy/error.go

```diff
@@ -65,39 +67,57 @@ func addErrs(e dErr, errs ...error) dErr {
 	return e
 }
 
-func errf(format string, a ...interface{}) dErr {
-	return newErr(fmt.Errorf(format, a...))
+// Errf returns a DError by constructing error message with given format.
+func Errf(format string, a ...interface{}) DError {
+	return newErr(format, fmt.Errorf(format, a...))
 }
 
-// newErr returns a dErr. newErr is used to wrap another error as a dErr.
-// If e is already a dErr, e is copied and returned.
+// newErr returns a DError. newErr is used to wrap another error as a DError.
+// If e is already a DError, e is copied and returned.
+// If e is a normal error, replacementMsg is used to hide privacy info.
 // If e is nil, nil is returned.
-func newErr(e error) dErr {
+func newErr(replacementMsg string, e error) DError {
 	if e == nil {
 		return nil
 	}
 	if dE, ok := e.(*dErrImpl); ok {
 		return dE
 	}
-	return &dErrImpl{errs: []error{e}}
+	return &dErrImpl{errs: []error{e}, errFormats: []string{replacementMsg}}
 }
 
-func typedErr(errType string, e error) dErr {
+// ToDError returns a DError. ToDError is used to wrap another error as a DError.
+// If e is already a DError, e is copied and returned.
+// If e is a normal error, error message is reused as format.
+// If e is nil, nil is returned.
+func ToDError(e error) DError {
 	if e == nil {
 		return nil
 	}
-	dE := newErr(e)
+	if dE, ok := e.(*dErrImpl); ok {
+		return dE
+	}
+	return &dErrImpl{errs: []error{e}, errFormats: []string{e.Error()}}
+}
+
+func typedErr(errType string, safeErrMsg string, e error) DError {
+	if e == nil {
+		return nil
+	}
+	safeErrMsg = fmt.Sprintf("%v: %v", errType, safeErrMsg)
+	dE := newErr(safeErrMsg, e)
 	dE.(*dErrImpl).errType = errType
 	return dE
 }
 
-func typedErrf(errType, format string, a ...interface{}) dErr {
-	return typedErr(errType, fmt.Errorf(format, a...))
+func typedErrf(errType, format string, a ...interface{}) DError {
+	return typedErr(errType, format, fmt.Errorf(format, a...))
 }
 
 type dErrImpl struct {
-	errs    []error
-	errType string
+	errs       []error
+	errFormats []string
```

Comment:
As mentioned in other comments, rename this to e.g. anonymizedErrs, as this can hold other values except formats (when newErr is called directly)